python/cudf/cudf/core/window/ewm.py

```diff
@@ -7,13 +7,46 @@
 
 import numpy as np
 
+import pylibcudf as plc
+
 from cudf.core.window.rolling import _RollingBase
 from cudf.utils.dtypes import is_dtype_obj_numeric
 
 if TYPE_CHECKING:
     from cudf.core.column.column import ColumnBase
 
 
+def _leading_nulls_count(plc_col: plc.Column, size: int) -> int:
+    if size == 0 or plc_col.null_count() == 0:
+        return 0
+
+    bitmask = plc_col.null_mask()
+    if bitmask is None:
+        return 0
+
+    offset = plc_col.offset()
+
+    total = plc.null_mask.null_count(bitmask, offset, offset + size)
+    if total == 0:
+        return 0
+
+    # binary search for the first first occurrence
+    # where null count == index
+    low = 0
+    high = size
+    while low + 1 < high:
+        midpoint = low + ((high - low) // 2)
+        null_count = plc.null_mask.null_count(
+            bitmask, offset, offset + midpoint
+        )
+        if null_count == midpoint:
+            low = midpoint
+        else:
+            high = midpoint
```

Comment:
This seems very inefficient to me. Each call to `plc.null_mask.null_count()` will launch a CUDA kernel.
I could create a new libcudf/pylibcudf API to retrieve the index of the first non-null row:
```cpp
cudf::size_type index_of_first_valid_bit( bitmask_type const* bitmask,
                                          cudf::size_type start,
                                          cudf::size_type stop,
                                          rmm::cuda_stream_view stream);
```